vng_api_common/caching/etags.py

```diff
@@ -49,6 +74,18 @@ def calculate_etag(instance: models.Model) -> str:
     request.versioning_scheme = api_settings.DEFAULT_VERSIONING_CLASS()
 
     serializer = serializer_class(instance=instance, context={"request": request})
+    return serializer
+
+
+def calculate_etag(instance: models.Model) -> str:
+    """
+    Calculate the MD5 hash of a resource representation in the API.
+
+    The serializer for the model class is retrieved, and then used to construct
+    the representation of the instance. Then, the representation is rendered
+    to camelCase JSON, after which the MD5 hash is calculated of this result.
+    """
+    serializer = get_etag_serializer(instance)
```

Comment:
Sorry I don't understand how it reduces the number of DB queries. Does initiating of serializer with instance create additional DB queries?

Reply:
Yeah because serializer.data is cached on the serializer instance, so every fresh serializer instance does all the related field queries again (e.g. zaak.status, zaak.resultaat, zaak.kenmerken, zaak.eigenschappen...)

Reply:
Is it possible that if the instance was changed the `serializer.data` would be different from instance data because of this caching?

Reply:
if you make changes to `instance` after accessing `serializer.data`, yes, but you shouldn't do that :P 

you have to keep the intention of these packages in mind (audittrail, notifications, etag), which is to do *something* with the serialization result from the API action _just_ before the response is sent to the client. So, making changes to the instance that would affect the serializer output after the serializer output was generated would also be sending incorrect data to the client in the response. It's a risk, but it's really someone doing something wrong in the project then and/or getting their order of operations wrong.

Reply:
Thanks for the explanation